### ISSUE TITLE
Inconsistent VADER results

### ISSUE THREAD

**Issue by JEnduriumK** [nltk/nltk#3071]
Installing both NLTK (and the `vader_lexicon`) and the `vaderSentiment` thing [from PyPi](https://pypi.org/project/vaderSentiment/), which I believe is a mirror of [this version](https://github.com/cjhutto/vaderSentiment), and running the same segments of text (Tweets, really. Specifically [this one](https://twitter.com/AudiOfficial/status/1566063486320476163)) results in different results. 

```python
from nltk.sentiment.vader import SentimentIntensityAnalyzer as nltk_vader
from vaderSentiment.vaderSentiment import SentimentIntensityAnalyzer as raw_vader

def show_difference(test_string:str):
    print(f'    (NLTK) {test_string}:\n{nltk_vader().polarity_scores(test_string)}')
    print(f'(Not-NLTK) {test_string}:\n{raw_vader().polarity_scores(test_string)}\n')

if __name__ == '__main__':
    show_difference('Not all heroes wear capes. Prepare for performance with (super) power, this is the fully electric Audi e-tron S. Discover more at https://t.co/WoOrQWM6Vp. #Audi #eMobility #etronS #FutureIsAnAttitude https://t.co/mBVOYF3cqi')
    show_difference('Not all heroes wear capes.')
    show_difference('Prepare for performance with (super) power, this is the fully electric Audi e-tron S. Discover more at https://t.co/WoOrQWM6Vp. #Audi #eMobility #etronS #FutureIsAnAttitude https://t.co/mBVOYF3cqi')
    show_difference('Prepare for performance with (super) power, this is the fully electric Audi e-tron S.')
    show_difference('Not all heroes wear capes. Prepare for performance with (super) power, this is the fully electric Audi e-tron S. Discover more at')
    show_difference('Not all heroes wear capes. Prepare for performance with (super) power, this is the fully electric Audi e-tron S. Discover more at . #Audi #eMobility #etronS #FutureIsAnAttitude')
    show_difference('Not all heroes wear capes. Prepare for performance with (super) power, this is the fully electric Audi e-tron S. Discover more at https://t.co/WoOrQWM6Vp.  https://t.co/mBVOYF3cqi')
```

```
    (NLTK) Not all heroes wear capes. Prepare for performance with (super) power, this is the fully electric Audi e-tron S. Discover more at https://t.co/WoOrQWM6Vp. #Audi #eMobility #etronS #FutureIsAnAttitude https://t.co/mBVOYF3cqi:
{'neg': 0.091, 'neu': 0.909, 'pos': 0.0, 'compound': -0.4023}
(Not-NLTK) Not all heroes wear capes. Prepare for performance with (super) power, this is the fully electric Audi e-tron S. Discover more at https://t.co/WoOrQWM6Vp. #Audi #eMobility #etronS #FutureIsAnAttitude https://t.co/mBVOYF3cqi:
{'neg': 0.083, 'neu': 0.797, 'pos': 0.12, 'compound': 0.2955}

    (NLTK) Not all heroes wear capes.:
{'neg': 0.403, 'neu': 0.597, 'pos': 0.0, 'compound': -0.4023}
(Not-NLTK) Not all heroes wear capes.:
{'neg': 0.403, 'neu': 0.597, 'pos': 0.0, 'compound': -0.4023}

    (NLTK) Prepare for performance with (super) power, this is the fully electric Audi e-tron S. Discover more at https://t.co/WoOrQWM6Vp. #Audi #eMobility #etronS #FutureIsAnAttitude https://t.co/mBVOYF3cqi:
{'neg': 0.0, 'neu': 1.0, 'pos': 0.0, 'compound': 0.0}
(Not-NLTK) Prepare for performance with (super) power, this is the fully electric Audi e-tron S. Discover more at https://t.co/WoOrQWM6Vp. #Audi #eMobility #etronS #FutureIsAnAttitude https://t.co/mBVOYF3cqi:
{'neg': 0.0, 'neu': 0.849, 'pos': 0.151, 'compound': 0.5994}

    (NLTK) Prepare for performance with (super) power, this is the fully electric Audi e-tron S.:
{'neg': 0.0, 'neu': 1.0, 'pos': 0.0, 'compound': 0.0}
(Not-NLTK) Prepare for performance with (super) power, this is the fully electric Audi e-tron S.:
{'neg': 0.0, 'neu': 0.769, 'pos': 0.231, 'compound': 0.5994}

    (NLTK) Not all heroes wear capes. Prepare for performance with (super) power, this is the fully electric Audi e-tron S. Discover more at:
{'neg': 0.114, 'neu': 0.886, 'pos': 0.0, 'compound': -0.4023}
(Not-NLTK) Not all heroes wear capes. Prepare for performance with (super) power, this is the fully electric Audi e-tron S. Discover more at:
{'neg': 0.102, 'neu': 0.752, 'pos': 0.147, 'compound': 0.2955}

    (NLTK) Not all heroes wear capes. Prepare for performance with (super) power, this is the fully electric Audi e-tron S. Discover more at . #Audi #eMobility #etronS #FutureIsAnAttitude:
{'neg': 0.098, 'neu': 0.902, 'pos': 0.0, 'compound': -0.4023}
(Not-NLTK) Not all heroes wear capes. Prepare for performance with (super) power, this is the fully electric Audi e-tron S. Discover more at . #Audi #eMobility #etronS #FutureIsAnAttitude:
{'neg': 0.086, 'neu': 0.791, 'pos': 0.123, 'compound': 0.2955}

    (NLTK) Not all heroes wear capes. Prepare for performance with (super) power, this is the fully electric Audi e-tron S. Discover more at https://t.co/WoOrQWM6Vp.  https://t.co/mBVOYF3cqi:
{'neg': 0.105, 'neu': 0.895, 'pos': 0.0, 'compound': -0.4023}
(Not-NLTK) Not all heroes wear capes. Prepare for performance with (super) power, this is the fully electric Audi e-tron S. Discover more at https://t.co/WoOrQWM6Vp.  https://t.co/mBVOYF3cqi:
{'neg': 0.094, 'neu': 0.769, 'pos': 0.136, 'compound': 0.2955}
```

Notably, the NLTK version seems to be far less responsive to the non-hashtag, non-url text content changing. For example, despite massive differences in the first two chunks of text, the NLTK version comes up with the exact same compound score, but the non-NLTK version comes up with different scores. (Which is why I'm reporting the issue here, thinking it might be an NLTK issue. Though it could be me somehow using this incorrectly?)

EDIT to show the results in a different, maybe more readable, way:
```
Text 1: Not all heroes wear capes. Prepare for performance with (super) power, this is the fully electric Audi e-tron S. Discover more at https://t.co/WoOrQWM6Vp. #Audi #eMobility #etronS #FutureIsAnAttitude https://t.co/mBVOYF3cqi
Text 2: Not all heroes wear capes.
Text 3: Prepare for performance with (super) power, this is the fully electric Audi e-tron S. Discover more at https://t.co/WoOrQWM6Vp. #Audi #eMobility #etronS #FutureIsAnAttitude https://t.co/mBVOYF3cqi
Text 4: Prepare for performance with (super) power, this is the fully electric Audi e-tron S.
Text 6: Not all heroes wear capes. Prepare for performance with (super) power, this is the fully electric Audi e-tron S. Discover more at
Text 7: Not all heroes wear capes. Prepare for performance with (super) power, this is the fully electric Audi e-tron S. Discover more at . #Audi #eMobility #etronS #FutureIsAnAttitude
Text 8: Not all heroes wear capes. Prepare for performance with (super) power, this is the fully electric Audi e-tron S. Discover more at https://t.co/WoOrQWM6Vp.  https://t.co/mBVOYF3cqi

    (NLTK) {'neg': 0.091, 'neu': 0.909, 'pos': 0.0, 'compound': -0.4023}
    (NLTK) {'neg': 0.403, 'neu': 0.597, 'pos': 0.0, 'compound': -0.4023}
    (NLTK) {'neg': 0.0,   'neu': 1.0,   'pos': 0.0, 'compound': 0.0}
    (NLTK) {'neg': 0.0,   'neu': 1.0,   'pos': 0.0, 'compound': 0.0}
    (NLTK) {'neg': 0.114, 'neu': 0.886, 'pos': 0.0, 'compound': -0.4023}
    (NLTK) {'neg': 0.098, 'neu': 0.902, 'pos': 0.0, 'compound': -0.4023}
    (NLTK) {'neg': 0.105, 'neu': 0.895, 'pos': 0.0, 'compound': -0.4023}

(Not-NLTK) {'neg': 0.083, 'neu': 0.797, 'pos': 0.12, 'compound': 0.2955}
(Not-NLTK) {'neg': 0.403, 'neu': 0.597, 'pos': 0.0, 'compound': -0.4023}
(Not-NLTK) {'neg': 0.0, 'neu': 0.849, 'pos': 0.151, 'compound': 0.5994}
(Not-NLTK) {'neg': 0.0, 'neu': 0.769, 'pos': 0.231, 'compound': 0.5994}
(Not-NLTK) {'neg': 0.102, 'neu': 0.752, 'pos': 0.147, 'compound': 0.2955}
(Not-NLTK) {'neg': 0.086, 'neu': 0.791, 'pos': 0.123, 'compound': 0.2955}
(Not-NLTK) {'neg': 0.094, 'neu': 0.769, 'pos': 0.136, 'compound': 0.2955}
```

**Comment by tomaarsen** [comment]
One of the differences that I'm noting is that NLTK uses:
```python
['Not', 'all', 'heroes', 'wear', 'capes', 'Prepare', 'for', 'performance', 'with', '(super)', 'power', 'this', 'is', 'the', 'fully', 'electric', 'Audi', 'e-tron', 'S.', 'Discover', 'more', 'at', 'https://t.co/WoOrQWM6Vp.', '#Audi', '#eMobility', '#etronS', '#FutureIsAnAttitude', 'https://t.co/mBVOYF3cqi']
```
while vaderSentiment uses:
```python
['Not', 'all', 'heroes', 'wear', 'capes', 'Prepare', 'for', 'performance', 'with', 'super', 'power', 'this', 'is', 'the', 'fully', 'electric', 'Audi', 'e-tron', 'S.', 'Discover', 'more', 'at', 'https://t.co/WoOrQWM6Vp', 'Audi', 'eMobility', 'etronS', 'FutureIsAnAttitude', 'https://t.co/mBVOYF3cqi']
```

These are equivalent except for two differences:
1. NLTK uses `(super)` while vaderSentiment uses `super`
2. NLTK uses `#Audi`, `#etronS` etc. while vaderSentiment uses `Audi`, `etronS`, etc.

The former issue can be explained by https://github.com/cjhutto/vaderSentiment/pull/57, which updates some punctuation differences. In particular, [this section](https://github.com/cjhutto/vaderSentiment/pull/57/files#diff-9a2748b116a0e4c9a01cdacfa3e16c05c087903d4c2953468b34770325effc61R181-R186) in vaderSentiment versus this section in NLTK:
https://github.com/nltk/nltk/blob/2f34b52da55a6c83fd18306095c9aa00b4996381/nltk/sentiment/vader.py#L299-L311



In fact, replacing `(super)` with `super` makes the outputs (almost) identical again:
```
    (NLTK) Not all heroes wear capes. Prepare for performance with super power, this is the fully electric Audi e-tron S. Discover more at https://t.co/WoOrQWM6Vp. Audi eMobility etronS FutureIsAnAttitude https://t.co/mBVOYF3cqi:
{'neg': 0.083, 'neu': 0.797, 'pos': 0.12, 'compound': 0.2955}
(Not-NLTK) Not all heroes wear capes. Prepare for performance with super power, this is the fully electric Audi e-tron S. Discover more at https://t.co/WoOrQWM6Vp. Audi eMobility etronS FutureIsAnAttitude https://t.co/mBVOYF3cqi:
{'neg': 0.083, 'neu': 0.797, 'pos': 0.12, 'compound': 0.2955}

    (NLTK) Not all heroes wear capes.:
{'neg': 0.403, 'neu': 0.597, 'pos': 0.0, 'compound': -0.4023}
(Not-NLTK) Not all heroes wear capes.:
{'neg': 0.403, 'neu': 0.597, 'pos': 0.0, 'compound': -0.4023}

    (NLTK) Prepare for performance with super power, this is the fully electric Audi e-tron S. Discover more at https://t.co/WoOrQWM6Vp. #Audi #eMobility #etronS #FutureIsAnAttitude https://t.co/mBVOYF3cqi:
{'neg': 0.0, 'neu': 0.849, 'pos': 0.151, 'compound': 0.5994}
(Not-NLTK) Prepare for performance with super power, this is the fully electric Audi e-tron S. Discover more at https://t.co/WoOrQWM6Vp. #Audi #eMobility #etronS #FutureIsAnAttitude https://t.co/mBVOYF3cqi:
{'neg': 0.0, 'neu': 0.849, 'pos': 0.151, 'compound': 0.5994}

    (NLTK) Prepare for performance with super power, this is the fully electric Audi e-tron S.:
{'neg': 0.0, 'neu': 0.769, 'pos': 0.231, 'compound': 0.5994}
(Not-NLTK) Prepare for performance with super power, this is the fully electric Audi e-tron S.:
{'neg': 0.0, 'neu': 0.769, 'pos': 0.231, 'compound': 0.5994}

    (NLTK) Not all heroes wear capes. Prepare for performance with super power, this is the fully electric Audi e-tron S. Discover more at:
{'neg': 0.102, 'neu': 0.752, 'pos': 0.147, 'compound': 0.2955}
(Not-NLTK) Not all heroes wear capes. Prepare for performance with super power, this is the fully electric Audi e-tron S. Discover more at:
{'neg': 0.102, 'neu': 0.752, 'pos': 0.147, 'compound': 0.2955}

    (NLTK) Not all heroes wear capes. Prepare for performance with super power, this is the fully electric Audi e-tron S. Discover more at . #Audi #eMobility #etronS #FutureIsAnAttitude:
{'neg': 0.088, 'neu': 0.784, 'pos': 0.127, 'compound': 0.2955}
(Not-NLTK) Not all heroes wear capes. Prepare for performance with super power, this is the fully electric Audi e-tron S. Discover more at . #Audi #eMobility #etronS #FutureIsAnAttitude:
{'neg': 0.086, 'neu': 0.791, 'pos': 0.123, 'compound': 0.2955}
# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
# The neg and pos here are the *only* differences between NLTK and vaderSentiment now.

    (NLTK) Not all heroes wear capes. Prepare for performance with super power, this is the fully electric Audi e-tron S. Discover more at https://t.co/WoOrQWM6Vp.  https://t.co/mBVOYF3cqi:        
{'neg': 0.094, 'neu': 0.769, 'pos': 0.136, 'compound': 0.2955}
(Not-NLTK) Not all heroes wear capes. Prepare for performance with super power, this is the fully electric Audi e-tron S. Discover more at https://t.co/WoOrQWM6Vp.  https://t.co/mBVOYF3cqi:
{'neg': 0.094, 'neu': 0.769, 'pos': 0.136, 'compound': 0.2955}
```

In conclusion, this is a difference in the text preprocessing between vaderSentiment and NLTK. I hope that helps explain it somewhat!

I don't think this is a bug per se, and I'm uncertain whether this is something that should be changed.
